src/ProductCard.tsx

```diff
@@ -0,0 +1,121 @@
+import classNames from 'classnames';
+import React, { useContext } from 'react';
+import { Link } from 'react-router-dom';
+import { Context } from './Context';
+import { Product } from './types/Product';
+import { LocaleStorageTypes } from './types/LocaleStorageTypes';
+import {
+  findProductOnCart,
+  findProductOnFavourites,
+  updateCart,
+  updateFavourites,
+} from './utils/productManipulations';
+
+type Props = {
+  product: Product | null,
+  favouritesTimeout?: number
+};
+
+export const ProductCard: React.FC<Props> = ({
+  product,
+  favouritesTimeout,
+}) => {
+  const {
+    setChosenProducts,
+    setProductsToBuy,
+    setLoadingItem,
+  } = useContext(Context);
+
+  return (
+    product && (
+      <Link
+        className="product__link"
+        to={`/${product.type}s/${product.id}`}
+        onClick={() => {
+          localStorage.setItem(
+            LocaleStorageTypes.product, JSON.stringify(product),
+          );
+        }}
+      >
+        {product && (
+          <div className="product">
+            <img className="product__image" src={product.imageUrl.split('phones').join('products')} alt={`${product.id}`} />
+            <h3 className="product__title">{product.name}</h3>
+            <div className="product__price_container">
+              {product.discount === 0 ? (
+                <div className="product__price">{`$${product.price}`}</div>
+              ) : (
+                <>
+                  <div className="product__price">{`$${product.price - (product.price * product.discount) / 100}`}</div>
+                  <div className="product__price product__price--old">{`$${product.price}`}</div>
+                </>
+              )}
+            </div>
+            <div className="product__line" />
+            <div className="product__info_container">
+              <div className="product__info">Screen</div>
+              <div className="product__info--value">
+                {product.screen || '-'}
+              </div>
+            </div>
+            <div className="product__info_container">
+              <div className="product__info">Capacity</div>
+              <div className="product__info--value">
+                {product.capacity || '-'}
+              </div>
+            </div>
+            <div
+              className="product__info_container product__info_container--last"
+            >
+              <div className="product__info">RAM</div>
+              <div className="product__info--value">{product.ram || '-'}</div>
+            </div>
+            <div className="product__button_container">
+              <button
+                type="button"
+                className={classNames(
+                  'product__button_add',
+                  {
+                    'product__button_add--active':
+                    findProductOnCart(product.id) === true,
```

Comment:
let's try to find the product somewhere in component body and save the result in var to avoid redundant function call